aksolve/test/aksolve_test.py

```diff
@@ -0,0 +1,187 @@
+
+   
+        ans = aks.cg(
+            matvec,
+            b,
+            x_start=ak.array([2, 1]),
+            tol=EPS,
+            max_iter=5,
+            verbose=True)
+        self.assertTrue(ak.all(ak.abs(ans[0] - x) < 1.0e-4))
+        self.assertLess(ans[3], EPS)
+        self.assertEqual(ans[2], 3)
+        self.assertEqual(ans[1], 2)
+
+        P = lambda x: ak.array([1/4, 1/3]) * x
+        ans = qks.cg(matvec, b, precon=P, verbose=True)
+        self.assertTrue(ak.all(ak.abs(ans[0] - x) < 1.0e-4))
+        self.assertLess(ans[3], EPS)
+        self.assertEqual(ans[2], 2)
+        self.assertEqual(ans[1], 2)
+
+
+        D = ak.randint(1, 1000, 2 ** 32, dtype='float64')
+        matvec = lambda x: x * D
+        x = ak.randint(0, 1, 2 ** 32, dtype='float64')
+        P = lambda x: x * (1.0 / D)
+        b = matvec(x)
+        ans = aks.cg(matvec, b, precon=P, verbose=True)
+        self.assertTrue(ak.all(ak.abs(ans[0] - x) < 1.0e-4))
+        self.assertLess(ans[3], 1.0e-6)
+        self.assertEqual(ans[2], 1)
+        self.assertEqual(ans[1], 3)
+
+
+    #minres tests
+    def test_MINRES(self):
+        # A = [[ 6.,  3.,  0.],
+        #      [ 3., -2.,  5.],
+        #      [ 0.,  5.,  2.]]
+        R = ak.array([0, 0, 0, 1,  1, 1, 2, 2, 2])
+        C = ak.array([0, 1, 2, 0,  1, 2, 2, 1, 2])
+        V = ak.array([6, 3, 0, 3, -2, 5, 0, 5, 2])
+
+        x = ak.array([2., -2., 9.])
+        b = ak.array([6., 55., 8.])
+
+        matvec = matvec_from_coo(R, C, V)
+        self.assertTrue(ak.all(matvec(x) == b))
+        
+
+        ans = aks.minres(matvec, ak.zeros(3), verbose=True)
+        self.assertEqual(ans[1], 0)
+
+        ans = aks.minres(matvec, b, x, verbose=True)
+        self.assertEqual(ans[1], 0)
+        self.assertTrue(ak.all(ans[0] == x))
+
+        ans = aks.minres(matvec, b, max_iter=3, verbose=True)
+        self.assertTrue(ak.all(ak.abs(ans[0] - x) < 1.0e-4))
+        self.assertLess(ans[3], 1.0e-6)
+        self.assertEqual(ans[2], 3)
+        self.assertEqual(ans[1], 1)
+
+    
+        invV = ak.array([0.15104167, 0.03125, -0.078125, 0.03125, -0.0625,
+                         0.15625, -0.078125, 0.15625, 0.109375])
+        precon = matvec_from_coo(R, C, invV)
+        ans = aks.minres(matvec, b, precon=precon, verbose=True)
+    
+        self.assertTrue(ak.all(ak.abs(ans[0] - x) < 1.0e-4))
+        self.assertLess(ans[3], 1.0e-6)
+        self.assertEqual(ans[2], 2)
+        self.assertEqual(ans[1], 2)
+
+        D = ak.randint(1, 1000, 2 ** 32, dtype='float64')
+        matvec = lambda x: x * D
+        x = ak.randint(0, 1, 2 ** 32, dtype='float64')
+        P = lambda x: x * (1.0 / D)
+        b = matvec(x)
+        ans = aks.minres(matvec, b, precon=P, verbose=True)
+        self.assertTrue(ak.all(ak.abs(ans[0] - x) < 1.0e-4))
+        self.assertLess(ans[3], 1.0e-6)
+        self.assertEqual(ans[2], 1)
+        self.assertEqual(ans[1], 3)
+
+
+#TODO: finish converting these
```

Comment:
do these things still need to be worked?

Reply:
Added issue #34 to track